shared/settings/advanced/index.tsx

```diff
@@ -208,6 +250,152 @@ class Developer extends React.Component<Props, State> {
   }
 }
 
+// A list so the order of the elements is fixed
+const ProxyTypeToDisplayNameList = [
+  ['noProxy', 'No Proxy'],
+  ['httpConnect', 'HTTP Connect'],
+  ['socks', 'Socks5'],
+]
+const ProxyTypeToDisplayName = ProxyTypeToDisplayNameList.reduce((obj, item) => {
+  obj[item[0]] = item[1]
+  return obj
+}, {})
+const DisplayNameToProxyType = {
+  'HTTP Connect': 'httpConnect',
+  'No Proxy': 'noProxy',
+  Socks5: 'socks',
+}
+
+type ProxyState = {
+  address: string
+  port: string
+  proxyType: string
+  certPinning: boolean
+  showCertPinningConfirmationModal: boolean
+}
+
+type ProxyProps = Props & {confirmDisableCertPinning: (disableCertPinning: () => void) => void}
+
+class ProxySettings extends React.Component<ProxyProps, ProxyState> {
+  constructor(props: ProxyProps) {
+    super(props)
+
+    this.state = {
+      address: '',
+      certPinning: true,
+      port: '',
+      proxyType: ProxyTypeToDisplayName['noProxy'],
+      showCertPinningConfirmationModal: false,
+    }
+  }
+
+  componentWillReceiveProps(nextProps: ProxyProps) {
+    if (nextProps.proxyData !== this.props.proxyData) {
+      var addressPort = nextProps.proxyData.addressWithPort.split(':')
+      var address = addressPort[0]
+      var port = '80'
+      if (addressPort.length >= 2) {
+        port = addressPort[1]
+      }
+
+      var certPinning = nextProps.proxyData.certPinning
+
+      var proxyType = ProxyTypeToDisplayName[RPCTypes.ProxyType[nextProps.proxyData.proxyType]]
+
+      this.setState({address, certPinning, port, proxyType})
+    }
+  }
+
+  handleAddressChange = (address: string) => {
```

Comment:
we don't need functions like this, just call setstate

Reply:
Done.